### ISSUE TITLE
clean mechanism for python project build files under it's src directory

### ISSUE THREAD

**Issue by lamikr** [lamikr/rocm_sdk_builder#57]
Python projects pytorch, pytorch_audio, pytorch_vision, onnxruntime, DeepSpeed, torch_migraphx and bitsandbytes does not build cleanly under builddir but instead under src-directory.

When doing a clean build, files under "src_projects/pytorch/build" for example does not get cleaned up and can prevent project reconfiguration if patches or other dependencies have changed.

Need to have a way to clean them properly and I was first thinking that there may be need for new hook-call for python. 

Ended up easier solution where each of these projects should have preconfig_rocm.sh script that can be called on project preconfig time. This way it's only called once and for each clean build. And if it is wanted to call again, user should remove it's build dir. (for example builddir/039_01_pytorch)

**Comment by daniandtheweb** [comment]
I still haven't tried it but doesn't git clean -fdx work?

**Comment by lamikr** [comment]
I did't know about that command! For now I called "python setup.py clean" for 2 or 3 of those 5 python projects which seemed to have somekind of support for it. For rest I checked the existence of build dir and then deleted it.

But "git -fdx" could be even better if it works. In pytorch I have seen situation when creating some patch to .cu file, that the hip file generated from it did not get updated just by cleaning the build directory.  (On beginning of pytorch build there are always executed hipify commands that does that)
